uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/staff-leave-report.js

```diff
@@ -156,11 +159,17 @@ define([
 
     // Init block
     (function init () {
-      $q.all([
-        loadStatuses(),
-        loadAbsenceTypes(),
-        loadAbsencePeriods()
-      ])
+      checkPermissions(sharedSettings.permissions.admin.administer)
+      .then(function (isAdmin) {
```

Comment:
I think you are trying to set the role, but we are already setting the role here https://github.com/civicrm/civihr/blob/d1c444dd03881aa21fd4af1b6d367d7ce808190c/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-container.js#L15-L15

Then we already know that we are in admin role using that rootscope variable. So I am not sure why checkPermission needed to be added here. And using the same rootScope variable we are deciding the role in other components, specially in leave-request-popup.